File: C++/BFS/BFS (adjacency matrix).cpp
#include<iostream>
#include<queue>
using namespace std;

bool visited[15];
bool adj_mtrx[15][15];
queue<int> q;
queue<int> ans;

void adj_check()
{
    // node 1 and 2 are connected
    adj_mtrx[1][2] = true;
    adj_mtrx[2][1] = true;
    
    // node 1 and 6 are connected
    adj_mtrx[1][6] = true;
    adj_mtrx[6][1] = true;
    
    // node 1 and 7 are connected
    adj_mtrx[1][7] = true;
    adj_mtrx[7][1] = true;
    
    // node 1 and 9 are connected
    adj_mtrx[1][9] = true;
    adj_mtrx[9][1] = true;
    
    // node 2 and 3 are connected
    adj_mtrx[2][3] = true;
    adj_mtrx[3][2] = true;
    
    // node 2 and 4 are connected
    adj_mtrx[2][4] = true;
    adj_mtrx[4][2] = true;
    
    // node 4 and 5 are connected
    adj_mtrx[4][5] = true;
    adj_mtrx[5][4] = true;
    
    // node 7 and 8 are connected
    adj_mtrx[7][8] = true;
    adj_mtrx[8][7] = true;
}

void bfs(int num)
{
    visited[num] = true;
    q.push(num);
    
    while(!q.empty())
    {
        int cur = q.front(); q.pop();
        visited[cur] = true;
        //cout << "->"<<cur;
        ans.push(cur);
        for(int i=1; i<10; i++)
        {
            if(!visited[i] && adj_mtrx[cur][i])
                q.push(i);
        }
        
    }
}

void ans_print()
{
    cout << "bfs search order\n";
    cout << ans.front(); ans.pop();
    while(!ans.empty())
    {
        int cur = ans.front(); ans.pop();
        cout << "->"<<cur;
    }
    cout <<"\n";
}

int main()
{
    adj_check();
    bfs(1);
    ans_print();
    exit(0);
}

File: Python/Graph/BFS/BFS (adjacency matrix).py
from collections import deque


# adjacency matrix
def adj_check(n, m):
    # node n and m are connected
    adj[n][m] = True
    adj[m][n] = True


def bfs(num):
    global adj, q, ans

    q.append(num)

    while len(q) > 0:
        cur = q.popleft()
        visited[cur] = True
        ans.append(cur)

        for i in range(1, 10):
            if adj[cur][i] is True and not visited[i]:
                q.append(i)


def ans_print():
    global ans
    print("bfs search order")
    temp = ans.popleft()
    print(temp, end="")

    while len(ans) > 0:
        print("->{}".format(ans.popleft()), end="")


if __name__ == '__main__':
    visited = [False for _ in range(15)]
    adj = [([False] * 10) for _ in range(10)]
    q = deque()
    ans = deque()

    adj_check(1, 2)
    adj_check(1, 6)
    adj_check(1, 7)
    adj_check(1, 9)

    adj_check(2, 3)
    adj_check(2, 4)

    adj_check(4, 5)

    adj_check(7, 8)

    bfs(1)
    ans_print()
